core/src/main/java/com/scalar/db/api/ScanAll.java

```diff
@@ -88,15 +88,18 @@ public ScanAll withEnd(Key clusteringKey, boolean inclusive) {
   }
 
   /**
-   * Guaranteed to throw an exception.
+   * Sets the specified scan ordering. Ordering can only be specified with arbitrary columns. To
+   * sort results by multiple columns, call this method multiple times in the order of sorting.
    *
-   * @throws UnsupportedOperationException always
-   * @deprecated Unsupported operation.
+   * @param ordering a scan ordering
+   * @return this object
+   * @deprecated As of release 3.6.0. Will be removed in release 5.0.0. Use the setter method of the
+   *     Scan builder instead; to create a Scan builder, use {@link Scan#newBuilder()}
    */
   @Deprecated
   @Override
   public ScanAll withOrdering(Ordering ordering) {
-    throw new UnsupportedOperationException();
+    return (ScanAll) super.withOrdering(ordering);
```

Comment:
Unlike a current ScanAll, a relational scan accept orderings. So, now we need to implement this, although we would like to use this internally for the builder and users do not have to use this.

Reply:
It's not guaranteed that `super.withOrdering()` returns ScanAll in terms of the type signature and it seems an unsafe downcast to me. I'm wondering if any type check is needed (or applying Generics type parameter, but it takes effort...) ?

Reply:
I thought it was safe because the caller object of `super.withOrdering()` here is always `ScanAll` (and we are already doing the same in other similar methods).
@brfrn169 What do you think?

Reply:
Yeah, it seems there is no reasonable solution to it. I think we can go with the implementation as is.

I wish Java had a special generics type `THIS` or something which represents the concrete class type itself so that we could write the method like `public THIS withOrdering()`...

Reply:
@brfrn169 @komamitsu Thank you! I left this as is.